### ISSUE TITLE
Improve error message of S1118, and it should not report issues if the class has members that use an instance of a class

### ISSUE THREAD

**Issue by tvsonar** [SonarSource/sonarlint-vs#388]
Error message: say explicitely that the constructor should be hidden by making it protected or private.

Like operator overloads, properties with the type of the class. These require the class to be instantiated, so the class can't be marked static.
Also, in
```
public class X
{
  public static void M(X x) {}
}
```
parameter `x` needs to be instantiated somewhere, so based on the visibility of the class this might be outside the class, which means that we can even propose a private constructor. So it is safer to just not report on these methods.
